CHAID/tree.py

```diff
@@ -303,6 +303,32 @@ def node_predictions(self):
                 pred[node.indices] = node.node_id
         return pred
 
+    def classification_rules(self, tree=None, node=None, stack=None):
+        if tree is None:
```

Comment:
I think I might re-write so it is bottom up instead of top down so we don't need the tree at all. Memotisation seems overkill just to get rid of an internal parameter.